### DISCUSSION TITLE
Performance when using compound variants

### DISCUSSION THREAD

**Discussion by afzalsayed96** [stitchesjs/stitches#271] [Ideas] [answered]
Hoping to continue the discussion from [discord](https://discord.com/channels/752614004387610674/752795957674115092/763793684448935977) here so others could chime in.

**Problem Description**

How far can we go with compoundVariants before we see a noticeable performance drop? [Here's](https://codesandbox.io/s/youthful-wind-i6r1u?file=/src/App.tsx) an example that creates `1296` compoundVariants for a component rendered many times. 

Should the responsibility of keeping number of compoundVariants lower than a certain threshold lie on the user? Or can the library add optimizations to the logic so user doesn't have to care about this?

**Snapshot from React profiler**

![Screen Shot 2020-10-08 at 9 27 52 PM](https://user-images.githubusercontent.com/14029371/95567467-9b065400-0a40-11eb-8146-f83ff4e7746f.png)

**Codesandbox Link**

https://codesandbox.io/s/youthful-wind-i6r1u?file=/src/App.tsx

**Comment by afzalsayed96** [comment]
Was thinking about this issue and just want to check if anything has cleared up on the topic.
To be clear we have run into this problem in a real codebase and anyone generating variants from theme tokens will do as well.

I think it's acceptable if there's a one-time cost on component creation but this seems to happen on every rerender which is a bit of a concern for me. Wondering if we could do some optimisations to change things 🤔 

cc @jonathantneal @peduarte

**Comment by peduarte** [comment, marked as answer]
We're aware of this and are re-working how the compound variant API works. Rest assured this will be fixed in the stable release 🔥
